AutomatedTesting/Gem/PythonTests/assetpipeline/scene_tests_fixtures/scene_test_debug_compare.py

```diff
@@ -0,0 +1,241 @@
+                                                      actual_file_path: str, expected_file_path: str) -> (List[str], List[str]):
+    """
+    This function will truncate any floating point values to be short enough to be stable.
+
+    o3de/o3de#13713: We need to stabilize the output of product assets, so this will emit a warning.
+
+    diff_actual: A list containing the debug output (dbgsg file) of the processed source asset under test.
+    diff_expected: A list containing the expected debug output of the source asset under test.
+    actual_file_path: Path to the dbgsg file containing the debug output of the processed source asset under test.
+    expected_file_path: Path to the dbgsg file containing the expected debug output to compare against.
+    """
+
+    # Get the initial list lengths, so they can be compared to the list lengths later to see if any differences were
+    # removed due to floating point value drift.
+    initial_diff_actual_len = len(diff_actual)
+    initial_diff_expected_len = len(diff_expected)
+
+    # Decide the number of digits past the decimal point to preserve.
+    DECIMAL_DIGITS_TO_PRESERVE = 3
+
+    # This function requires the two lists to be equal length.
+    assert initial_diff_actual_len == initial_diff_expected_len, "Scene mismatch - different line counts"
+
+    # Store the trimmed values and compare the un-trimmed values separately, emitting warnings.
+    # Trim decimals from the lists to be compared, if any where found, re-compare and generate new lists.
+    floating_point_regex = re.compile(
+        f"(.*?-?[0-9]+\\.[0-9]{{{DECIMAL_DIGITS_TO_PRESERVE},{DECIMAL_DIGITS_TO_PRESERVE}}})[0-9]+(.*)")
+    for index, diff_actual_line in enumerate(diff_actual):
+        # Loop, because there may be multiple floats on the same line.
+        while True:
+            match_result = floating_point_regex.match(diff_actual[index])
+            if not match_result:
+                break
+            diff_actual[index] = f"{match_result.group(1)}{match_result.group(2)}"
+        # diff_actual and diff_expected have the same line count, so they can both be checked here
+        while True:
+            match_result = floating_point_regex.match(diff_expected[index])
+            if not match_result:
+                break
+            diff_expected[index] = f"{match_result.group(1)}{match_result.group(2)}"
+
+    # Re-run the diff now that floating point values have been truncated.
+    diff_actual, diff_expected = utils.get_differences_between_lists(diff_actual, diff_expected)
+
+    # If both lists are now empty, then the only differences between the two scene files were floating point drift.
+    if (diff_actual == None and diff_expected == None) or (len(diff_actual) == 0 and len(diff_actual) == 0):
+        warnings.warn(f"Floating point drift detected between {expected_file_path} and {actual_file_path}.")
```

Comment:
May worth checking whether the original diff_actual and diff_expected passed to this function are None/empty since you may not want to throw this warning if there's nothing to compare from the beginning.

Reply:
Added to issue list for phase 2.